File: LeetCode/array/MissingNumber.java
package array;

public class MissingNumber {
    public int missingNumber(int[] nums) {

        int n = nums.length;
        int sum = (n * (n + 1)) / 2;
        int cur = 0;

        for (int num: nums) {
            cur += num;
        }

        return sum - cur;
    }

    public static void main(String[] args) {
        MissingNumber solution = new MissingNumber();

        // Test cases
        int[] nums1 = {3, 0, 1};
        int[] nums2 = {0, 1};
        int[] nums3 = {9, 6, 4, 2, 3, 5, 7, 0, 1};

        System.out.println("Missing number in nums1: " + solution.missingNumber(nums1)); // Output: 2
        System.out.println("Missing number in nums2: " + solution.missingNumber(nums2)); // Output: 2
        System.out.println("Missing number in nums3: " + solution.missingNumber(nums3)); // Output: 8
    }
}

File: LeetCode_Python/math/MissingNumber.py
class MissingNumber:
    def missingNumber(self, nums: list[int]) -> int:
        n = len(nums)
        total = n * (n + 1) // 2

        for num in nums:
            total -= num
        return total


# Time Complexity: O(n)
# Space Complexity: O(1)


if __name__ == "__main__":
    solution = MissingNumber()

    print("Expected Output: 2")
    print("Actual Output:", solution.missingNumber([3, 0, 1]))

    print("Expected Output: 8")
    print("Actual Output:", solution.missingNumber([0, 1, 2, 3, 4, 5, 6, 7, 9]))
